src/Gui/DlgGeneralImp.cpp

```diff
@@ -215,6 +215,25 @@ void DlgGeneralImp::saveSettings()
     hGrp->GetGroup("TreeView")->SetBool("Enabled",treeView);
     hGrp->GetGroup("PropertyView")->SetBool("Enabled",propertyView);
 
+    //save workbench selector position
+    hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/MainWindow");
+    bool WSToolbar = false, WSLeftCorner = false, WSRightCorner = false;

```

Comment:
This whole logic is weird because the 3 boolean are actually mutual exclusive but the way how it's done here they can be combined because each parameter can be set to true or false independently (directly in the user.cfg).
This results into the weird behaviour that if the keys WSLeftCorner and WSRightCorner are set to true two switcher appear, one on the left side and that in the toolbar (not that on the right side)

The three parameters should be replaced by one string parameter called WSPosition and its values are "WSToolbar", "WSLeftCorner" and "WSRightCorner". The default or in case of an invalid value "WSToolbar" is used.

Reply:
Thanks.
I understand.
So this + change the connect to the new style. Is that all?
Do you want to push directly or should I make a PR for those changes?

Reply:
@wwmayer how about using a `PrefComboBox`? This class today stores the index of the selected item, which is a bit boring because then in the code it's like dealing with magic numbers. But it exists already and at some point we can maybe improve it to store a user-defined value instead of index.

Reply:
@0penBrain the difficulty with PrefComboBox is that the string of an item cannot be used because this usually is translated text and changing the language will break the reading of the parameter. The alternative is to use a combo box' user data but in most cases a programmer doesn't set it.

I think the best we can do is defining an enum and a map or array of QString with the translated text. The index of the combo box is cast to the enum.

@PaddleStroke 
> So this + change the connect to the new style. Is that all? Do you want to push directly or should I make a PR for those changes?

Yes, that's it. And yes I will have a look...

Reply:
See #7744